haystack/document_stores/pinecone.py

```diff
@@ -0,0 +1,682 @@
+                "match the number of embeddings in Pinecone. Make sure your Pinecone "
+                "index aligns to the same database that was used when creating the "
+                "original index."
+            )
+
+    def write_documents(
+        self,
+        documents: Union[List[dict], List[Document]],
+        index: Optional[str] = None,
+        batch_size: int = 32,
+        duplicate_documents: Optional[str] = None,
+        headers: Optional[Dict[str, str]] = None,
+    ) -> None:
+        """
+        Add new documents to the DocumentStore.
+
+        :param documents: List of `Dicts` or List of `Documents`. If they already contain the embeddings, we'll index
+                          them right away in Pinecone. If not, you can later call update_embeddings() to create & index them.
+        :param index: (SQL) index name for storing the docs and metadata
+        :param batch_size: When working with large number of documents, batching can help reduce memory footprint.
+        :param duplicate_documents: Handle duplicates document based on parameter options.
+                                    Parameter options : ( 'skip','overwrite','fail')
+                                    skip: Ignore the duplicates documents
+                                    overwrite: Update any existing documents with the same ID when adding documents.
+                                    fail: an error is raised if the document ID of the document being added already
+                                    exists.
+        :raises DuplicateDocumentError: Exception trigger on duplicate document
+        :return: None
+        """
+        if headers:
+            raise NotImplementedError("PineconeDocumentStore does not support headers.")
+
+        index = index or self.index
+        index = self._sanitize_index_name(index)
+        duplicate_documents = duplicate_documents or self.duplicate_documents
+        assert (
+            duplicate_documents in self.duplicate_documents_options
+        ), f"duplicate_documents parameter must be {', '.join(self.duplicate_documents_options)}"
+
+        if not self.pinecone_indexes.get(index):
+            self.pinecone_indexes[index] = self._create_index_if_not_exist(
+                vector_dim=self.vector_dim,
+                index=index,
+                metric_type=self.metric_type,
+                replicas=self.replicas,
+                shards=self.shards,
+            )
+
+        field_map = self._create_document_field_map()
+        document_objects = [Document.from_dict(d, field_map=field_map) if isinstance(d, dict) else d for d in documents]
+        document_objects = self._handle_duplicate_documents(
+            documents=document_objects, index=index, duplicate_documents=duplicate_documents
+        )
+        if len(document_objects) > 0:
+            add_vectors = False if document_objects[0].embedding is None else True
+            # I don't think below is required
```

Comment:
Not sure what you are referring to with this comment..

Reply:
I think I had already removed the code, but missed the comment - now I removed the comment